### ISSUE TITLE
[MA0134] False positive when using null conditional operator

### ISSUE THREAD

**Issue by Pretasoc** [meziantou/Meziantou.Analyzer#512]
### Version of the Meziantou.Analyzer NuGet package

2.0.43

### Rule Identifier

MA0134

### Target Framework

net6.0

### C# Language version

C# 11

### Description

When an async method is invoked using a null conditional operator a warning is emitted, although the result ist discarded as described in the documentation.

### Reproduction Steps

Minimal code:

```c#
public interface IFoo
{
    Task BarAsync();
}

public static void Baz(IFoo? foo)
{
    _ = foo?.BarAsync();
}
```

**Comment by meziantou** [comment]
If `foo` is not null, the task is executed, but the result is not used. So, I'm not sure this is a false-positive.

Should your code be:

````c#
if (foo != null)
{
   await foo.BarAsync();
}
````

**Comment by Pretasoc** [comment]
Thats not the reason, why i'm thinking, its a false positive. In your example you said `_ = Task.Delay(1);` is ok. I am aware, that a task maybe is executed, but i don't care for the result, if it is. At least, thats what i'm trying to express with discard.

Of course i could write:

````
if (foo != null)
{
   _ = foo.BarAsync();
}
```

which would also not raise a warning. But i find this rather complex, compared to the single line avobe.

**Comment by meziantou** [comment]
I see... I think you are right!

The expected behavior is:

````c#
// report diagnostic
instance.ReturnTask();
instance?.ReturnTask();

// ok
_ = instance.ReturnTask();
_ = instance?.ReturnTask();
````
